api/v1beta1/foundationdbcluster_types.go

```diff
@@ -2893,6 +2901,17 @@ type RoutingConfig struct {
 	// This feature is only supported in FDB 7.0 or later, and requires
 	// dual-stack support in your Kubernetes environment.
 	PodIPFamily *int `json:"podIPFamily,omitempty"`
+
+	// UseDNSInClusterFile determines whether to use DNS names rather than IP
+	// addresses to identify coordinators in the cluster file.
+	// NOTE: This is an experimental feature, and is not supported in the
+	// latest stable version of FoundationDB.
+	UseDNSInClusterFile *bool `json:"useDNSInClusterFile,omitempty"`
+
+	// DNSSuffix defines the cluster domain used in a DNS name generated for a
+	// service.
+	// The default is `cluster.local`.
+	DNSDomain *string `json:"dnsSuffix,omitempty"`
```

Comment:
Can we add some validation marker here like the default, a max length e.g.:
```suggestion
	DNSDomain *string `json:"domain,omitempty"`
```

We can also call if `dnsDomain`.

```
+kubebuilder:validation:MinLength=1
+kubebuilder:validation:MaxLength=253 
```

and perhaps we want to add a `+kubebuilder:validation:Pattern="..."`we should take the DNS validation regex from Kubernetes to match the expectations: https://github.com/kubernetes/apimachinery/blob/release-1.23/pkg/util/validation/validation.go#199-L200 the cluster behaviour. We might also want to use some of those methods in our code later, if we have to do some validation at a place.

Reply:
I see the benefit in adding the validation on the pattern, but the pattern required is pretty messy, so I'd rather not embed it in the CRD. We may want to consider adding that logic in the code if it can make errors with the DNS Domain format more visible.